client/my-sites/checkout/modal/index.tsx

```diff
@@ -70,8 +75,12 @@ const CheckoutModal: FunctionComponent< Props > = ( {
 		checkoutOnSuccessCallback?.();
 		onClose?.();
 
-		// Reload the page to get latest data
-		window.location.href = redirectTo;
+		if ( isRelativeUrl( redirectTo ) ) {
+			// Using full page reload here to refresh the latest plan data for the new site
+			window.location.href = redirectTo;
+		} else {
+			window.location.href = previousRouteWithArgs;
```

Comment:
In the case of the assembler flow, editing the redirect_to param to something like example.com results in using this branch but the previousRouteWithArgs value is set to the current url so the checkout completes payment and then reloads the same checkout again.

I'm not sure if this is an acceptable failure mode, it might be better to just bail out to `navigate('/');` instead. (The fallback is applied above anyway, the only concern here is that the redirect target is something we control / can't be set by url param. (previous route is stored in state and should be safe enough)

Reply:
>  previousRouteWithArgs value is set to the current url so the checkout completes payment and then reloads the same checkout again.

I'm not an expert on the checkout modal (it was [made by another team](https://github.com/Automattic/wp-calypso/pull/60181)) but will this reload checkout or will it reload the gutenberg page that we're actually on? It seems like the purpose of this redirect is just to reload the editor so whatever feature you just purchased is made available.

Reply:
It may be that if `redirectTo` is not a relative URL, something is very wrong. If that's the case, we should log in logstash and display an error to the user.